### PR TITLE
remove self and bind

### PR DESCRIPTION
**Pull request by LinusU**
This removes the use of `self` and `bind`, and replaces it with the awesome fat arrow (`=>`).

I have also updated some of the functions where there was a lot of unnecessary wrapping with `new Promise`, but not all of the places. I wanted this pull request to focus on only `self` and `bind`.

I have also marked three `FIXME`s in the code, again it didn't seem right to fix it in this pull request; and in some cases I wanted the original author to maybe clarify the intent. I think we should merge with them in, and then fix them later.

I would appreciate a fast pull here since this will probably need manual rebasing as soon as anything else lands :)

ping @johnnyman727, I saw that you where online, would you like to review? Thanks :+1: 

edit: also, I need to go to bed right away, sorry about that, I'll check this first thing in the morning!

edit2: note to self to update this comment when landing, https://github.com/tessel/t2-cli/issues/470#issuecomment-160171309

### CODE REVIEW

**Comment by LinusU** [comment]
Argh, one of the test depended on Zalgo :laughing: 

Looking into it now...

**Comment by LinusU** [comment]
Should work now :+1:

**Comment by johnnyman727** [comment]
@LinusU will check out today!

**Comment by rwaldron** [comment]
I'm a little torn on this one. It's really too broad of a change set for one PR. It will also cause existing PRs that are more focused to produce merge conflicts, but those will and first and so this will require merge conflict resolution. I'd much prefer, for everyone's sanity sake, that these changes be broken out into different changesets, with different PRs.

File: lib/tessel/provision.js [hunk 1]
```diff
@@ -106,15 +98,13 @@ actions.setupLocal = function(keyFile) {
...
-          fs.writeFile.bind(this, keyFile + '.pub', publicKey, {
-            encoding: 'utf8',
-            mode: 384
-          }),
-          fs.writeFile.bind(this, keyFile, privateKey, {
-            encoding: 'utf8',
-            mode: 384
-          }),
+          (cb) => fs.writeFile(keyFile + '.pub', publicKey, fileOptions, cb),
+          (cb) => fs.writeFile(keyFile, privateKey, fileOptions, cb),
```
Comment (rwaldron) [inline lib/tessel/provision.js]: So much nicer.

**Comment by johnnyman727** [comment]
@rwaldron which PRs do you want to get merged before this one?

**Comment by LinusU** [comment]
@rwaldron When thinking of it, I agree with you, it should probably have been multiple PRs. I'll be sure to rebase as soon as other patches land though, and then we'll get away with it :)

**Comment by LinusU** [comment]
I just rebased on the latest master.

@rwaldron where there any more pull requests you wanted merged before this one?

**Comment by LinusU** [comment]
Rebased on master again

nudge nudge @rwaldron ;)

**Comment by rwaldron** [comment]
@LinusU the branch I want in hasn't landed.

**Comment by rwaldron** [comment]
To be as honest as possible: I'm being selfish because I don't want to resolve conflicts in deploy.js for the sake of a non-essential patch :P

**Comment by rwaldron** [comment]
Looks like there was no conflict—sorry for being annoying about that :|

**Comment by johnnyman727** [comment]
By the way, @LinusU this PR is super impressive. :+1:

**Comment by LinusU** [comment]
No problem, thanks for the merge!

Thank you :)

**Pull request merged**